File: alerts/notification-service/internal/domain/event/class_invitation/user_details_responded_event.go
package classinvitation

import (
	"notification-serivce/internal/domain/dto"
	"notification-serivce/internal/domain/enums"
	"notification-serivce/internal/domain/models"
	"reflect"
)

type UserDetails struct {
	dto.UserDTO
	Role string `json:"role"`
}

type UserDetailsRespondedEvent struct {
	ID       string      `json:"notificationId"`
	Sender   UserDetails `json:"sender"`
	Receiver UserDetails `json:"receiver"`
}

func (u *UserDetailsRespondedEvent) Name() string {
	return reflect.TypeOf(u).Elem().Name()
}

func (u *UserDetailsRespondedEvent) FieldsToUpdate() map[string]any {
	return map[string]any{
		"receiver": *models.NewUser(u.Receiver.ID, u.Receiver.FirstName, u.Receiver.LastName, u.Receiver.Role),
		"sender":   *models.NewUser(u.Sender.ID, u.Sender.FirstName, u.Sender.LastName, u.Sender.Role),
		"status":   enums.CREATED,
	}
}

File: account/user-service/src/main/java/org/tutora/userservice/event/UserDetailsRespondedEvent.java
package org.tutora.userservice.event;

import java.util.List;

public record UserDetailsRespondedEvent(
        String notificationId,
        UserDetails sender,
        UserDetails receiver
) implements Event {

    @Override
    public String name() {
        return this.getClass().getSimpleName();
    }

    public record UserDetails(
            String id,
            String firstName,
            String lastName,
            List<String> roles
    ) {}
}
